data/trie/patriciaMerkleTrie.go

```diff
@@ -515,6 +556,24 @@ func (tr *patriciaMerkleTrie) Snapshot() error {
 	return nil
 }
 
+func (tr *patriciaMerkleTrie) newSnapshotDb() (data.DBWriteCacher, error) {
+	db, err := storageUnit.NewDB(
+		storageUnit.DBType(tr.snapshotDbCfg.Type),
+		path.Join(tr.snapshotDbCfg.FilePath, strconv.Itoa(tr.snapshotId)),
+		tr.snapshotDbCfg.BatchDelaySeconds,
+		tr.snapshotDbCfg.MaxBatchSize,
+		tr.snapshotDbCfg.MaxOpenFiles,
```

Comment:
Warning: we can reach the 1024 opened files limit by using this mechanism if checkpoints are done too often.

Reply:
The only case in which the Checkpoint creates a new db, is if there is no snapshot db (see L469). This is the case if the Checkpoint is called, and no Snapshot was called ever before. Only `Snapshot()` creates new db, so only one new db will be created each epoch.

Reply:
Changed the code a bit, so if Checkpoint is called and no Snapshot was called before, Checkpoint will call Snapshot.